src/model.ts

```diff
@@ -690,9 +701,16 @@ export class Model extends EventBus<any> implements CommandDispatcher {
   }
 
   updateMode(mode: Mode) {
+    if (mode === "dashboard" && !(this.config.transportService instanceof LocalTransportService)) {
+      throw new Error(
+        "Cannot switch to dashboard mode in a collaborative session. The model shouldn't have a transport service."
+      );
+    }
     // @ts-ignore For testing purposes only
     this.config.mode = mode;
-    this.trigger("update");
+    if (this.status === Status.Ready) {
+      this.trigger("update");
+    }
```

Comment:
Probably need to change so we can change the mode of the model in a clean way.

And aren't all those `if (this.status === Status.Ready) {` pretty and cool ? 🙈 🙉